python_modules/dagster/dagster/components/utils.py

```diff
@@ -183,3 +183,45 @@ def get_path_from_module(module: ModuleType) -> Path:
         else None
     )
     return check.not_none(module_path, f"Module {module.__name__} has no filepath")
+
+
+def copy_fields_to_model(copy_from: type[BaseModel], copy_to: type[BaseModel]) -> type[BaseModel]:
```

Comment:
I feel like this name should communicate that its making a new class 

`create_model_subclass_with_fields_from` 
`create_shadow_model` 
`create_mirrored_model`